### DISCUSSION TITLE
script should execute on local when the target host is localhost or 127.0.0.1

### DISCUSSION THREAD

**Discussion by koolay** [umputun/spot#204] [General]
script should execute on local when the target host is localhost or 127.0.0.1.

**Comment by umputun** [comment]
Can you provide any context? I don't think the system should exhibit such behavior, as it would be surprising and unexpected. Often, I want to see the real SSH connection to the localhost when testing my playbooks. The current method of explicitly setting `local` seems more obvious and less confusing to me.

**Reply by koolay** [reply]
It do not need to remote ssh if spot run on one of the targets. And it do not need to set option on to local on every commond.

**Reply by umputun** [reply]
This tool is designed to run playbooks either via SSH or locally. Setting it to "local" mode eliminates the need for SSH transport, which seems preferable to implicitly modifying the transport to use local execution, as mentioned earlier.

Also, to add even more confusion - why just localhost or 127.0.0.1? How about `::1`, or `::ffff:127.0.0.1`? And what about the LAN IP of the box? Or the entire range designated for localhost, i.e., `127.0.0.0/8`? As you can see, this is not as obvious as it sounds, and it will add confusion and complexity for no good reason.

**Reply by koolay** [reply]
There is a option for local on host in `Ansiable`. It is a global option and not only task or command level option.
![image](https://github.com/umputun/spot/assets/4273490/3accc11f-847e-4ee6-b4e3-6b4523500a43)
